<h1> Palmer Penguin Data Analytics </h1>
<hr>
<h3> Analyst: Inciong, Avian Cayle </h3>
<h4> BSIT-3A </h4>

In [39]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p>Importing dataset to Jupyter notebook</p>
<p>Converting dataset to dataframe</p>
<ul><b>Requriements</b>
    <li>Name of the dataframe</li>
    <li>Source</li>
</ul>

<h4> Descriptive Analytics </h4>

In [40]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [41]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

In [42]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h4> Cleaning the Data Frame </h4>

In [6]:
penguins.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [7]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

In [8]:
penguins.isnull().diff()

species island bill_length_mm bill_depth_mm flipper_length_mm body_mass_g  \
0       NaN    NaN            NaN           NaN               NaN         NaN   
1     False  False          False         False             False       False   
2     False  False          False         False             False       False   
3     False  False           True          True              True        True   
4     False  False           True          True              True        True   
..      ...    ...            ...           ...               ...         ...   
339   False  False          False         False             False       False   
340   False  False          False         False             False       False   
341   False  False          False         False             False       False   
342   False  False          False         False             False       False   
343   False  False          False         False             False       False   

       sex   year  
0      NaN    NaN  
1    False  False  
2    False  False  
3     True  False  
4     True  False  
..     ...    ...  
339  False  False  
340  False  False  
341  False  False  
342  False  False  
343  False  False  

[344 rows x 8 columns]

<h3> Selecting a column </h3>
<h4> Syntax </h4>
data_frame[column_name]

In [9]:
[penguins['island'] == 'Dream']

[0      False
 1      False
 2      False
 3      False
 4      False
        ...  
 339     True
 340     True
 341     True
 342     True
 343     True
 Name: island, Length: 344, dtype: bool]

In [10]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [11]:
x = penguins[penguins['species'] == 'Gentoo']
y = penguins[penguins['species'] == 'Adelie']
z = penguins[penguins['species'] == 'Chinstrap']

<h3> GENTOO SPECIES </h3>

In [12]:
x

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

<h3> ADELIE SPECIES </h3>

In [13]:
y

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

<h3> CHINSTRAP SPECIES </h3>

In [14]:
z

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

<h3> HOW MANY CHINSTRAP HAS THE BODY MASS OVER 4000? </h3>

In [15]:
massOfChinstrap = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000)]

In [16]:
massOfChinstrap.info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [17]:
sexOfChinstrap = penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000) & (penguins['sex'] == 'male')]

In [18]:
sexOfChinstrap.info()

<class 'pandas.core.frame.DataFrame'>
Index: 15 entries, 289 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            15 non-null     object 
 1   island             15 non-null     object 
 2   bill_length_mm     15 non-null     float64
 3   bill_depth_mm      15 non-null     float64
 4   flipper_length_mm  15 non-null     float64
 5   body_mass_g        15 non-null     float64
 6   sex                15 non-null     object 
 7   year               15 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h3> HOW MANY CHINSTRAP PENGUINS IS MALE? </h3>

In [19]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000) & (penguins['sex'] == 'male')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 15 entries, 289 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            15 non-null     object 
 1   island             15 non-null     object 
 2   bill_length_mm     15 non-null     float64
 3   bill_depth_mm      15 non-null     float64
 4   flipper_length_mm  15 non-null     float64
 5   body_mass_g        15 non-null     float64
 6   sex                15 non-null     object 
 7   year               15 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h3> HOW MANY CHINSTRAP PENGUINS IS FEMALE? </h3>

In [20]:
penguins[(penguins['species'] == 'Chinstrap') & (penguins['body_mass_g'] >= 4000) & (penguins['sex'] == 'female')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 1 entries, 284 to 284
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            1 non-null      object 
 1   island             1 non-null      object 
 2   bill_length_mm     1 non-null      float64
 3   bill_depth_mm      1 non-null      float64
 4   flipper_length_mm  1 non-null      float64
 5   body_mass_g        1 non-null      float64
 6   sex                1 non-null      object 
 7   year               1 non-null      int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 72.0+ bytes


<h3> HOW TO CHANGE THE VALUE AS A NULL? </h3>

In [21]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [22]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

In [23]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')
penguins['year'] = penguins['year'].astype('category')

In [24]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


<h3> HOW TO CREATE A COLUMN </h3>

In [25]:
penguins['body_mass_kg'] = np.round(penguins['body_mass_g'] / 1000)
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen        39.10000       18.70000         181.000000   
1       Adelie  Torgersen        39.50000       17.40000         186.000000   
2       Adelie  Torgersen        40.30000       18.00000         195.000000   
3       Adelie  Torgersen        43.92193       17.15117         200.915205   
4       Adelie  Torgersen        36.70000       19.30000         193.000000   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream        55.80000       19.80000         207.000000   
340  Chinstrap      Dream        43.50000       18.10000         202.000000   
341  Chinstrap      Dream        49.60000       18.20000         193.000000   
342  Chinstrap      Dream        50.80000       19.00000         210.000000   
343  Chinstrap      Dream        50.20000       18.70000         198.000000   

     body_mass_g     sex  year  body_mass_kg  
0    3750.000000    male  2007           4.0  
1    3800.000000  female  2007           4.0  
2    3250.000000  female  2007           3.0  
3    4201.754386    male  2007           4.0  
4    3450.000000  female  2007           3.0  
..           ...     ...   ...           ...  
339  4000.000000    male  2009           4.0  
340  3400.000000  female  2009           3.0  
341  3775.000000    male  2009           4.0  
342  4100.000000    male  2009           4.0  
343  3775.000000  female  2009           4.0  

[344 rows x 9 columns]

<hr>

<h1> EXPLORATORY DATA ANALYTICS (EDA) </h1>

<h3> 1. What is the average body mass of male vs. female penguins? </h3>

In [26]:
penguins[penguins['sex'] == 'male']['body_mass_g'].mean()

np.float64(4514.684406547094)

In [27]:
penguins[penguins['sex'] == 'female']['body_mass_g'].mean()

np.float64(3862.2727272727275)

<h3> #1 INSIGHT </h3>

<h4> Male Penguins has the average body mass of 4545.68g, while female has 3862.27g body mass.</h4>

<hr>

<h3> 2. What is the most common sex for each species? </h3>

In [28]:
penguins.groupby(['sex', 'species']).size()

sex     species  
female  Adelie       73
        Chinstrap    34
        Gentoo       58
male    Adelie       79
        Chinstrap    34
        Gentoo       66
dtype: int64

<h3> #2 INSIGHT </h3>

<h4> Adelie penguins have an equal number of males and females, Chinstrap penguins also have equal numbers of both sexes, while Gentoo penguins have more males than females. </h4>

<hr>

<h3> 3. How many penguins were recorded each year? </h3>

In [29]:
penguins['year'].value_counts()

year
2009    120
2008    114
2007    110
Name: count, dtype: int64

<h3> #3 INSIGHT </h3>

<h4> In 2007, there's 110 penguins, in 2008 there's 114 and lastly, in 2009 there's 120 penguins. </h4>

<hr>

<h3> 4. How many penguins are on each island? </h3>

In [30]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h3> #4 INSIGHT </h3>

<h4> Biscoe has 168 penguins, dream has 124 and Torgersen has 52 penguins recorded. </h4>

<hr>

<h3> 5. How many male penguins are living in the Dream Island?</h3>

In [31]:
penguins[(penguins['island'] == 'Dream') & (penguins['sex'] == 'male')].value_counts().sum()

np.int64(63)

<h3> #5 INSIGHT </h3>

<h4> There are 62 male penguins living on Dream Island. </h4>

<hr>

<h3> 6. What is the average body mass of penguins per island? </h3>

In [32]:
penguins.groupby('island')['body_mass_g'].mean()

island
Biscoe       4712.956871
Dream        3712.903226
Torgersen    3715.899123
Name: body_mass_g, dtype: float64

<h3> #6 INSIGHT </h3>

<h4> Biscoe has the average of 4716g, Dream is 3712g while Torgersen is 3706g. </h4>

<hr>

<h3> 7. What is the summary of body_mass_g for all penguins? </h3>

In [33]:
penguins['body_mass_g'].sum()

np.float64(1445403.50877193)

<h3> #7 INSIGHT </h3>

<h4> The summary of body_mass_g for all penguins is 14370000. </h4>

<hr>

<h3> 8. How many penguins are there per sex, per island?</h3>

In [34]:
penguins.groupby(['sex', 'island']).size()

sex     island   
female  Biscoe       80
        Dream        61
        Torgersen    24
male    Biscoe       88
        Dream        63
        Torgersen    28
dtype: int64

<h3> #8 INSIGHT </h3>

<h4> There are 80 female and 83 male penguins on Biscoe, 61 female and 62 male on Dream, and 24 female and 23 male on Torgersen. </h4>

<hr>

<h3> 9. What is average flipper length per year and per species? </h3>

In [35]:
penguins.groupby(['year', 'species'])['flipper_length_mm'].mean()

year  species  
2007  Adelie       186.878304
      Chinstrap    192.423077
      Gentoo       215.117647
2008  Adelie       191.040000
      Chinstrap    197.722222
      Gentoo       217.565217
2009  Adelie       192.076923
      Chinstrap    198.083333
      Gentoo       218.020800
Name: flipper_length_mm, dtype: float64

<h3> #9 INSIGHT </h3>

<h4> In 2007, the average flipper length was 186.6 mm for Adelie, 192.4 mm for Chinstrap, and 215.1 mm for Gentoo. In 2008, it was 191.0 mm for Adelie, 197.7 mm for Chinstrap, and 217.6 mm for Gentoo. In 2009, it was 192.1 mm for Adelie, 198.1 mm for Chinstrap, and 218.4 mm for Gentoo. </h4>

<hr>

<h3> 10. What is the average bill depth by island and species? </h3>

In [36]:
penguins.groupby(['island', 'species'])['bill_depth_mm'].mean()

island     species  
Biscoe     Adelie       18.370455
           Chinstrap          NaN
           Gentoo       14.999606
Dream      Adelie       18.251786
           Chinstrap    18.420588
           Gentoo             NaN
Torgersen  Adelie       18.404830
           Chinstrap          NaN
           Gentoo             NaN
Name: bill_depth_mm, dtype: float64

<h3> #10 INSIGHT </h3>

<h4> The average bill depth is about 18.37 mm for Adelie and 14.98 mm for Gentoo on Biscoe, 18.25 mm for Adelie and 18.42 mm for Chinstrap on Dream, and 18.43 mm for Adelie on Torgersen. </h4>

<hr>

<h3> 11. What is the correlation between bill length and bill depth? </h3>

In [37]:
penguins['bill_length_mm'].corr(penguins['bill_depth_mm'])

np.float64(-0.23505287035553282)

<h3> #11 INSIGHT </h3>

<h4> The correlation between bill length and bill depth is approximately -0.24, showing a weak negative relationship. </h4>

<hr>

<h3> 12. What is the correlation between bill length and flipper length? </h3>

In [38]:
penguins['bill_length_mm'].corr(penguins['flipper_length_mm'])

np.float64(0.6561813407464278)

<h3> #12 INSIGHT </h3>

<h4> The correlation between bill length and flipper length is about 0.66, showing a moderate positive relationship. </h4>

<hr>